Lib/test/test_funcattrs.py

```diff
@@ -93,6 +93,20 @@ def f(): print(a)
             self.fail("shouldn't be able to read an empty cell")
         a = 12
 
+    def test_set_cell(self):
+        a = 12
+        def f(): print(a)
```

Comment:
If you use `def f(): return a` instead, you could verify that calling f() does return the updated value.